### PR TITLE
[Slack Adapter] Move methods from SlackAdapter class

### PR DESCRIPTION
**Pull request by matiasroldan6**
## Description
Moved methods that don't belong to the Slack Adapter class into either the API Wrapper or the Helper class, leaving the Adapter to deal with only the BotAdapter responsibilities.

**Methods moved from SlackAdapter:**

- LoginWithSlack
- GetBotUserByTeamAsync

**Properties moved from SlackAdapter:**

- Options 
- Identity

 

**Methods deleted:**

- GetInstallLink (not used)
- ValidateOauthCodeAsync (not used)
- GetAPIAsync (deprecated)

### CODE REVIEW

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs [hunk 1]
```diff
@@ -23,23 +23,17 @@ public class SlackAdapter : BotAdapter, IBotFrameworkHttpAdapter
...
         /// <param name="slackClient">An initialized instance of the SlackClientWrapper class.</param>
-        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
-        public SlackAdapter(SlackClientWrapper slackClient, SlackAdapterOptions options)
+        public SlackAdapter(SlackClientWrapper slackClient)
             : base()
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
-
-            if (string.IsNullOrWhiteSpace(_options.VerificationToken) && string.IsNullOrWhiteSpace(_options.ClientSigningSecret))
+            if (string.IsNullOrWhiteSpace(slackClient.Options.VerificationToken) && string.IsNullOrWhiteSpace(slackClient.Options.ClientSigningSecret))
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs]: I think this validation should be in SlackClientWrapper.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs [hunk 2]
```diff
@@ -592,5 +613,51 @@ public virtual async Task<FileUploadResponse> UploadFileAsync(byte[] fileData, s
         {
             return await _api.UploadFileAsync(fileData, fileName, channelIds, title, initialComment, useAsync, fileType).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Manages the login to Slack with the given credentials.
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A Task representing the asynchronous operation.</returns>
+        public async Task LoginWithSlack(CancellationToken cancellationToken)
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs]: This method should be called _LoginWithSlackAsync_

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs [hunk 3]
```diff
@@ -293,7 +216,7 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
                 return;
             }
 
-            if (!SlackHelper.VerifySignature(_options.ClientSigningSecret, request, body))
+            if (!SlackHelper.VerifySignature(_slackClient.Options.ClientSigningSecret, request, body))
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs]: Probably VerifySignature method belongs to the wrapper class too. It doesn't call any API method but it's totally related to it.

Comment (matiasroldan6) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs]: ok

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs [hunk 4]
```diff
@@ -293,7 +200,7 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
                 return;
             }
 
-            if (!SlackHelper.VerifySignature(_options.ClientSigningSecret, request, body))
+            if (!_slackClient.VerifySignature(_slackClient.Options.ClientSigningSecret, request, body))
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs]: The ClientSigningSecret is already in _slackClient, there's no need to pass it here.

Comment (matiasroldan6) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs]: my bad

**Pull request merged**